pkg/kubelet/server/server.go

```diff
@@ -209,12 +212,27 @@ func ListenAndServeKubeletReadOnlyServer(
 	}
 }
 
+type PodResourcesConfig struct {
+	MaxCallsPerSecond float64
+	MaxCallsBurst     int
+	Socket            string
+	PodsProvider      podresources.PodsProvider
+	DevicesProvider   podresources.DevicesProvider
+	CpusProvider      podresources.CPUsProvider
+	MemoryProvider    podresources.MemoryProvider
+}
+
```

Comment:
maybe you should have 2 structs
1. PodResourcesConfig (MaxCallsPerSecond , MaxCallsBurst, Socket)
2. PodResourcesProviders (PodsProvider, DevicesProvider   ...)

The PodResourcesProviders provders can be pass to the podresources.NewV1PodResourcesServer and podresources.NewV1alpha1PodResourcesServer so it will look like  podresources.NewV1PodResourcesServer(providers)

Reply:
I don't have strong feelings here and I can explore this direction